packages/server-wallet/src/utilities/validate-transition.ts

```diff
@@ -1,11 +1,32 @@
 import _ from 'lodash';
-import {SignedState, toNitroState} from '@statechannels/wallet-core';
+import {SignedState, StateWithHash, toNitroState} from '@statechannels/wallet-core';
 import {constants} from 'ethers';
+import {Transaction} from 'knex';
 
 import {logger} from '../logger';
 import {validateTransitionWithEVM} from '../evm-validator';
 import {Bytes} from '../type-aliases';
+import {Channel} from '../models/channel';
 
+export async function shouldValidateTransition(
+  incomingState: StateWithHash,
+  channel: Channel,
+  tx: Transaction
+): Promise<boolean> {
+  const {channelId} = channel;
+
+  // If we already have the state we should of already validated it
+  const alreadyHaveState = _.some(channel.sortedStates, ['stateHash', incomingState.stateHash]);
+
+  const {supported} = channel;
+
+  // Ignore older states that may be added via syncing
+  const isOldState = incomingState.turnNum < (supported?.turnNum || 0);
+
+  const isLedger = await Channel.isLedger(channelId, tx);
```

Comment:
I'd rather not couple `Channel` in a utility file if we can avoid it. We could parameterize this perhaps? You can check if it's a ledger channel by seeing if `channel.assetHolderAddress` is `undefined` or not

Reply:
Wait, why should we need to make a DB query to answer this question? (Should `Channel.isLedger` even exist as a method?)

Reply:
Good point, fixed in 09704fa0094a4e7b486a82a956c3933952dc523c

Reply:
> Wait, why should we need to make a DB query to answer this question? (Should `Channel.isLedger` even exist as a method?)

As far as I can tell it's only used by `Store.isLedger` which I don't think is used anywhere? So it probably should be removed.